### DISCUSSION TITLE
local build error

### DISCUSSION THREAD

**Discussion by 2239559319** [vercel/turborepo#7136] [Help] [answered]
error: failed to run custom build command for `turborepo-lib v0.1.0 (/home/user/desktop/personal/github/turbo/crates/turborepo-lib)`

Caused by:
  process didn't exit successfully: `/home/user/desktop/personal/github/turbo/target/debug/build/turborepo-lib-7fa3a2b0fd4da26a/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-changed=turbod.proto
  cargo:rerun-if-changed=../../cli/internal/turbodprotocol

  --- stderr
  thread 'main' panicked at crates/turborepo-lib/build.rs:33:28:
  tonic_build command: Custom { kind: Other, error: "protoc failed: turbod.proto:93:12: Explicit 'optional' labels are disallowed in the Proto3 syntax. To define 'optional' fields in Proto3, simply remove the 'optional' label, as fields are 'optional' by default.\n" }
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error: failed to run custom build command for `turbo v0.1.0 (/home/user/desktop/personal/github/turbo/crates/turborepo)`

Caused by:
  process didn't exit successfully: `/home/user/desktop/personal/github/turbo/target/debug/build/turbo-4d989fc8776e3471/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-changed=../../cli
  make[1]: Entering directory '/home/user/desktop/personal/github/turbo/cli'
  make protoc
  make[2]: Entering directory '/home/user/desktop/personal/github/turbo/cli'
  protoc --go_out=. --go_opt=paths=source_relative \
        --go-grpc_out=. --go-grpc_opt=paths=source_relative \
        internal/turbodprotocol/turbod.proto
  make[2]: Leaving directory '/home/user/desktop/personal/github/turbo/cli'
  make[1]: Leaving directory '/home/user/desktop/personal/github/turbo/cli'

  --- stderr
  internal/turbodprotocol/turbod.proto:93:12: Explicit 'optional' labels are disallowed in the Proto3 syntax. To define 'optional' fields in Proto3, simply remove the 'optional' label, as fields are 'optional' by default.
  make[2]: *** [Makefile:114: protoc] Error 1
  make[1]: *** [Makefile:119: internal/turbodprotocol/turbod.pb.go] Error 2
  thread 'main' panicked at crates/turborepo/build.rs:43:5:
  failed to build go binary
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
make: *** [Makefile:33: turbo] Error 101

**Comment by 2239559319** [comment]
how do i resolve this error?

**Reply by chris-olszewski** [reply, marked as answer]
I believe this is caused by an incorrect version of `protoc` can you make sure you're using a version `3.15` or later?

Also, it might be worth pulling from main again and trying to rebuild as it might have been an issue with just building Go which now no longer exists.

**Reply by 2239559319** [reply]
thanks. It works after updating protoc to 3.15.8
